pkg/bot/lark.go

```diff
@@ -0,0 +1,231 @@
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+package bot
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/larksuite/oapi-sdk-go/core"
+	"github.com/larksuite/oapi-sdk-go/core/tools"
+	"github.com/larksuite/oapi-sdk-go/event"
+	eventhttpserver "github.com/larksuite/oapi-sdk-go/event/http/native"
+
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/execute"
+	"github.com/infracloudio/botkube/pkg/log"
+	"github.com/infracloudio/botkube/pkg/utils"
+)
+
+const (
+	//LarkEvent lark event
+	LarkEvent = "event"
+	//ChatType lark chat type
+	ChatType = "chat_type"
+	//Group lark chat type
+	Group = "group"
+	//Text lark chat message
+	Text = "text_without_at_bot"
+	//OpenChatID lark chat id
+	OpenChatID = "open_chat_id"
+	//ChatID lark chat id
+	ChatID = "chat_id"
+	//OpenID lark user id
+	OpenID = "open_id"
+	//Users lark users
+	Users = "users"
+	//UserID lark user id
+	UserID = "user_id"
+	//Message eventType When sending a message to a chat group
+	Message = "message"
+	//AddBot eventType When a bot is added to a chat group
+	AddBot = "add_bot"
+	//P2pChatCreate eventType When a session is first created with the bot
+	P2pChatCreate = "p2p_chat_create"
+	//AddUserToChat eventType When a user is added to a chat group
+	AddUserToChat = "add_user_to_chat"
+	//LarkStartMsg lark start message
+	LarkStartMsg = "Hello from botkube~ Play with me by at botkube <commands>"
+)
+
+// LarkBot listens for user's message, execute commands and sends back the response
+type LarkBot struct {
+	AllowKubectl     bool
+	RestrictAccess   bool
+	ClusterName      string
+	DefaultNamespace string
+	Port             int
+	MessagePath      string
+	LarkClient       *utils.LarkClient
+}
+
+// NewLarkBot returns new Bot object
+func NewLarkBot(c *config.Config) Bot {
+	larkConf := c.Communications.Lark
+	appSettings := core.NewInternalAppSettings(core.SetAppCredentials(larkConf.AppID, larkConf.AppSecret),
+		core.SetAppEventKey(larkConf.VerificationToken, larkConf.EncryptKey))
+	conf := core.NewConfig(core.Domain(larkConf.Endpoint), appSettings, core.SetLoggerLevel(core.LoggerLevelError))
```

Comment:
Can we use log level from log pkg `github.com/infracloudio/botkube/pkg/log` `log.Level`

Reply:
@Lion916 this is not addressed yet